pkg/cloudprovider/suite_test.go

```diff
@@ -314,6 +323,104 @@ var _ = Describe("CloudProvider", func() {
 			Expect(err).ToNot(HaveOccurred())
 			Expect(isDrifted).To(BeTrue())
 		})
+		It("should return drifted if the AWSNodeTemplate securitygroup do not match the machine", func() {
```

Comment:
This checks the empty case IIUC, so maybe we should describe the test case as checking that

Reply:
AFAIK, this shouldn't return drifted right? https://github.com/aws/karpenter/pull/3949/files#diff-24b89fbce3fa5104d7328f74ecdfe7e57d275728825e337fa0307f94f8b071c3R99 says we'll return false.

Reply:
This is not checking the empty case. This is checking a normal drifted case